Inscription

Asper-01 opened the pull request:
Fichier débugué à MERGE.

luger95 commented on admin/headerAdmin.php:
```diff
@@ -0,0 +1,76 @@
+            <div class="collapse navbar-collapse" id="navbarNav">
+                <ul class="navbar-nav mr-auto">
+                   
+                    <?php if (userConnected()) { ?>
```
Dans les fichiers de vue, il est conseillé d'utiliser les syntaxes PHP adaptés.
```suggestion
                    <?php if (userConnected()): ?>
```

luger95 commented on admin/headerAdmin.php:
```diff
@@ -0,0 +1,76 @@
+                   
+                    <?php if (userConnected()) { ?>
+                    <?php } ?>
+                    <?php //var_dump($_SESSION);
+                    if (AdminConnected())
+                    //Fonction d'affichage du lien vers le CRUD (uniquement si l'utilisateur est Admin 1=Admin 0=User)+++++++++++++++++++++++++++++++   
+                    { ?>
```
```suggestion
                    <?php if (AdminConnected()): ?>
                    <?php //Fonction d'affichage du lien vers le CRUD (uniquement si l'utilisateur est Admin 1=Admin 0=User)  
                    ?>
```

luger95 commented on admin/headerAdmin.php:
```diff
@@ -0,0 +1,76 @@
+                        <li class="nav-item active">
+                            <a class="nav-link" href="userAdmin.php">Gestion Users<span class="sr-only">(current)</span></a>
+                        </li>
+                    <?php } ?>
```
```suggestion
                    <?php endif; ?>
```

luger95 commented on admin/headerAdmin.php:
```diff
@@ -0,0 +1,76 @@
+                    <?php } ?>
+                </ul>
+                <ul>
+                    <?php if (userConnected()) {
+                        //Fonction PhP pour afficher le boutton "Mon Compte" si l'utilisateur est connecté: $_SESSION['connexion'] = true; si le mdp et le log sont corrects dans connection.php 
+                    ?>
```
```suggestion
                    <?php if (userConnected()): ?>
                        <?php //Fonction PhP pour afficher le boutton "Mon Compte" si l'utilisateur est connecté 
                    ?>
```

luger95 commented on admin/headerAdmin.php:
```diff
@@ -0,0 +1,76 @@
+                                <li class="d-flex"><a href="../deconnexion.php" class="btn btn-danger">deconnexion</a></li>
+                            </div>
+                        </div>
+                    <?php } else { //Si non connecté on affiche "Connexion"
+                    ?>
```
```suggestion
                    <?php else: //Si non connecté on affiche "Connexion"
                    ?>
```

luger95 commented on admin/headerAdmin.php:
```diff
@@ -0,0 +1,76 @@
+                    <?php } else { //Si non connecté on affiche "Connexion"
+                    ?>
+                        <li class="d-flex"><a href="../form_co.php" class="btn btn-success">Connexion</a></li>
+                    <?php } ?>
```
```suggestion
                    <?php endif; ?>
```

luger95 commented on admin/productCreate.php:
```diff
@@ -0,0 +1,118 @@
+    $stmt->execute(["id" => $id]);
+    // On renvoie l'utilisateur à la page CRUD + message de confirmation
+    header('location:recettes.php');
+    $_SESSION['response'] = "Champ effacé de la base de donnée !";
+    $_SESSION['res_type'] = "danger";
```
Ce code ne sera pas exécuté, tu as une redirection au-dessus.

luger95 commented on admin/productCreate.php:
```diff
@@ -0,0 +1,118 @@
+    move_uploaded_file($_FILES['image']['tmp_name'], $upload);
+    // On renvoie l'utilisateur à la page CRUD + message de confirmation
+    header('location:/admin/product.php');
+    $_SESSION['response'] = "Insertion réussie en base de donnée !";
+    $_SESSION['res_type'] = "success";
```
Ce code ne sera pas exécuté, tu as une redirection au-dessus.

Asper-01 replied:
corrigé

luger95 commented on cart.php:
```diff
@@ -0,0 +1,133 @@
+
+
+// Initiatilaisation du panier dans la session
+if (isset($_SESSION['cart']) === false) {
+	$_SESSION['cart'] = [
+		'products' => [],
+		'total' => 0,
+	];
+}
```
Tu peux supprimer cette partie, tu ne l'utilises plus.

luger95 commented on cart.php:
```diff
@@ -0,0 +1,133 @@
+	return $total;
+}
+
+// function setProductInCart(array $query, int $qty)
+// {
+// 	$productId = $query['id'];
+
+// 	$_SESSION['cart']['products'][$productId] = [
+// 		'idProduct' => $query['id'],
+// 		'price' => $query['prix'],
+// 		'quantity' => $qty,
+// 	];
+
+// 	echo '<pre>';
+// 	var_dump($_SESSION);
+
+// 	calculTotalPriceCart();
+// }
```
à supprimer

luger95 commented on cart.php:
```diff
@@ -0,0 +1,133 @@
+}
+
+// Fonction de calcul du prix total panier
+function calculTotalPriceCart()
+{
+	$total = 0;
+	foreach ($_SESSION['cart']['products'] as $product) {
+		$total += intval($product['price']) * $product['quantity'];
+	}
+	$_SESSION['cart']['total'] = $total;
+	return $total;
+}
```
Si cette fonction n'est plus utilisé, il faut la supprimer

luger95 commented on cart.php:
```diff
@@ -0,0 +1,133 @@
+	if(!$quantity){
+		$quantity = 1;
+	}
+	//$userId = $_POST['id'];
```
à supprimer

luger95 commented on cart.php:
```diff
@@ -0,0 +1,133 @@
+			$stmt->execute(["id_product" => $pid, "user_id"=> $_SESSION["id"], 'quantity' => $quantity]); 
+		}
+
+	} header('location:/cart.php');
+	return true;
```
```suggestion
	}
	header('location:/cart.php');
```

luger95 commented on cart.php:
```diff
@@ -0,0 +1,133 @@
+	$stmt->execute(["id" => $id]);
+	// On renvoie l'utilisateur à la page CRUD + message de confirmation
+	header('location:cart.php');
+	$_SESSION['response'] = "Champ effacé de la base de donnée !";
+	$_SESSION['res_type'] = "danger";
```
Ce code ne sera pas éxécuté

luger95 commented on cart.php:
```diff
@@ -0,0 +1,133 @@
+$stmt->execute(["id" => $sid]);
+$query = $stmt->fetchAll();
+
+var_dump($_SESSION['id']);
```
à supprimer

luger95 commented on fonction.php:
```diff
@@ -31,6 +28,18 @@ function AdminConnected()
+ */
+function ClientConnected()
+{
+    if (userConnected() && $_SESSION['isAdmin'] === 0) {
+        return true;
+    }
+    return false;
```
```suggestion
    return (userConnected() && $_SESSION['isAdmin'] === 0);
```

luger95 commented on header.php:
```diff
@@ -21,66 +21,51 @@
+            <a class="navbar-brand" href="index.php">Junk Food</a>
             <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
                 <span class="navbar-toggler-icon"></span>
             </button>
             <div class="collapse navbar-collapse" id="navbarNav">
                 <ul class="navbar-nav mr-auto">
+                <?php if (ClientConnected()) { ?>
```
Utiliser une syntaxe adapté au contexte d'une page view

luger95 commented on header.php:
```diff
@@ -21,66 +21,51 @@
                     <?php } ?>
-                    <?php if (userConnected()) {
-                        //Fonction PhP pour afficher l'icône' "Mon panier" si l'utilisateur est connecté: $_SESSION['connexion'] = true; si le mdp et le log sont corrects dans connection.php 
+                    <?php if (userConnected()) {     
```
Utiliser une syntaxe adapté au contexte d'une page view

luger95 commented on index.php:
```diff
@@ -23,22 +23,25 @@
-                                <input type="hidden" class="pimage" value="<?= $row['preparation'] ?>">
-                                <input type="hidden" class="pcode" value="<?= $row['image'] ?>">
-                                <button class="btn btn-info btn-block addItemBtn"><i class="fas fa-cart-plus"></i>&nbsp;&nbsp;Ajouter au panier</button>
+                                <?php if (userConnected()) {
```
Utiliser une syntaxe adapté au contexte d'une page view

luger95 commented on index.php:
```diff
@@ -23,22 +23,25 @@
+                                    <input type="hidden" name="pid" class="pid" value="<?= $row['id'] ?>">
+                                    <button class="btn btn-info btn-block addItemBtn"><i class="fas fa-cart-plus"></i>&nbsp;&nbsp;Ajouter au panier</button>
 
+                                <?php } else if (!userConnected())  {  
```
Utiliser une syntaxe adapté au contexte d'une page view

luger95 commented on inscription_traitement.php:
```diff
@@ -1,65 +1,101 @@
-                        if (strlen($email) <= 45) { // On verifie que la longueur du mail <= 45
-                            if (filter_var($email, FILTER_VALIDATE_EMAIL)) { // Si l'email est de la bonne forme
-                                if ($password === $password_retype) { // si les deux mdp saisis sont bon
+                    if (!preg_match("#^[a-z0-9][a-z0-9]*[-]?[a-z0-9]*[a-z0-9]$#i", "{($pseudo)}")) {
+                        if (!empty($nom)) {
+                            if (!empty($prenom)) {
+                                if (strlen($email) <= 25) { // On vérifie que la longueur du mail <= 25
+                                    if (filter_var($email, FILTER_VALIDATE_EMAIL)) { // Si l'email est de la bonne forme
+                                        if (!empty($adresse)) {
+                                            if (!empty($code_postal)) {
+                                                if (!empty($ville)) {
+                                                    if ($password === $password_retype) { // Si les deux mots de passe saisis sont bons
```
Il faut revoir cette partie.
Ce genre de d'implémentation n'est pas une bonne pratique

luger95 reviewed (changes requested):
Des éléments sont à revoir

Asper-01 committed:
modif de: inscription.php inscription_traitement.php
-ajout champs formulaire et inscription en base
-ajout messages erreur
-il est de nouveau possible de s'inscrire
